src/backends/onnx.js

```diff
@@ -175,23 +175,27 @@ const ONNX_ENV = ONNX?.env;
 if (ONNX_ENV?.wasm) {
     // Initialize wasm backend with suitable default settings.
 
-    // (Optional) Set path to wasm files. This is needed when running in a web worker.
-    // https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#wasmPaths
-    // We use remote wasm files by default to make it easier for newer users.
-    // In practice, users should probably self-host the necessary .wasm files.
-    ONNX_ENV.wasm.wasmPaths = `https://cdn.jsdelivr.net/npm/@huggingface/transformers@${env.version}/dist/`;
+    // (Optional) Set path to wasm files. This will override the default path search behavior of onnxruntime-web.
+    // Currently, this behavior is disabled. To enable it, uncomment the following code.
+
+    // // Override the wasm search path if:
+    // // 1. We are not in a service worker.
+    // // 2. ONNX Runtime Web version is defined.
+    // // 3. The wasmPaths are not already set.
+    // //
+    // // @ts-ignore Cannot find name 'ServiceWorkerGlobalScope'.ts(2304)
+    // if (!(typeof ServiceWorkerGlobalScope !== 'undefined' && self instanceof ServiceWorkerGlobalScope)
+    //     && typeof ONNX?.versions?.web === 'string'
+    //     && !ONNX_ENV.wasm.wasmPaths) {
+    //     ONNX_ENV.wasm.wasmPaths = `https://cdn.jsdelivr.net/npm/onnxruntime-web@${ONNX.versions.web}/dist/`;
+    // }
```

Comment:
We've run into several issues in the past where the wasm files weren't able to be loaded locally, so we default to the CDN. Maybe in future we can switch to this, but I think we can keep it for now.

Another reason is that the raw .wasm file is ~21MB, but the CDN (gzipped) version is ~4MB. It might be a good idea to find a way to compress the raw file too. cc @fs-eire @guschmue

Reply:
No problem. I will make a change to keep them.

> We've run into several issues in the past where the wasm files weren't able to be loaded locally

This is indeed the case back to a few versions ago. I made it much better now but I still cannot say it's working for 100% cases. 

> the CDN (gzipped) version is ~4MB.

I think it is already the default setting for all modern server to enable gzip. A dev server usually does not enable this feature. However I didn't check the file size.